drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -657,43 +641,48 @@ function DragAndDropBlock(runtime, element, configuration) {
 
     var placeItem = function($zone, $item) {
         var item_id;
-        var $anchor;
         if ($item !== undefined) {
             item_id = $item.data('value');
-            // Element was placed using the mouse,
-            // so use relevant properties of *item* when calculating new position below.
-            $anchor = $item;
         } else {
             item_id = $selectedItem.data('value');
-            // Element was placed using the keyboard,
-            // so use relevant properties of *zone* when calculating new position below.
-            $anchor = $zone;
         }
+
         var zone = String($zone.data('uid'));
         var zone_id = $zone.data('zone_id');
         var zone_align = $zone.data('zone_align');
-        var $target_img = $root.find('.target-img');
 
-        // Calculate the position of the item to place relative to the image.
-        var x_pos = $anchor.offset().left + ($anchor.outerWidth()/2) - $target_img.offset().left;
-        var y_pos = $anchor.offset().top + ($anchor.outerHeight()/2) - $target_img.offset().top;
-        var x_pos_percent = x_pos / $target_img.width() * 100;
-        var y_pos_percent = y_pos / $target_img.height() * 100;
+        var items_in_zone_count = countItemsInZone(zone);
+        if (configuration.max_items_per_zone && configuration.max_items_per_zone <= items_in_zone_count) {
```

Comment:
@itsjeyd no, this code works before the item is placed, so if we allow it to continue when `items_in_zone_count == max_items_per_zone` it will place one more item and we will exceed the limit.